### PR TITLE
Fix/ms surface/ath10k firmware source

### LINKED ISSUE #288: Bad source for surface go wifi
https://github.com/NixOS/nixos-hardware/blob/41775780a0b6b32b3d32dcc32bb9bc6df809062d/microsoft/surface/firmware/surface-go/ath10k/ath10k-replace.nix#L5

^ this doesn't exist

https://github.com/NixOS/nixos-hardware/blob/41775780a0b6b32b3d32dcc32bb9bc6df809062d/microsoft/surface/repos.nix#L10-L15

https://github.com/kvalo/ath10k-firmware/tree/84b47062aab31d67156e0a3ef593a6999a12864b

### PR DESCRIPTION
**Pull request by mexisme**
The upstream firmware source had changed, which means the hash of the source + the patch needed to be updated.

### CODE REVIEW

File: microsoft/surface/firmware/surface-go/ath10k/ath10k-replace.nix [hunk 1]
```diff
@@ -22,7 +22,7 @@ firmwareLinuxNonfree.overrideAttrs (old: rec {
     cp ${killernetworking_firmware}/board.bin $out/lib/firmware/ath10k/QCA6174/hw3.0/
   '';
 
-  outputHash = "176cf5b9f370x2a532h2afjfkrxy13gqdygc11bam0sg8dlnrv21";
```
Comment (mexisme) [inline microsoft/surface/firmware/surface-go/ath10k/ath10k-replace.nix:25]: Fixes #288, @ursi ?

When I tested this, the above source was correct, but the hash after-patching needed to be updated

Comment (ursi) [reply microsoft/surface/firmware/surface-go/ath10k/ath10k-replace.nix:25]: unfortunately not
```
error: hash mismatch in fixed-output derivation '/nix/store/afk3yyi7apxdad0bbwd5fi5wf71m9qvs-microsoft-surface-go-firmware-linux-nonfree-20211027.drv':
         specified: sha256-UD0LzWOiLnlrNeiDyvqWq0FuNe5wjMNaV+hKHZLWzzk=
            got:    sha256-diTvvSrCBY4mDKwEwJkwDmWltYivNsCztf7NrPDXRx0=
```

Comment (mexisme) [reply microsoft/surface/firmware/surface-go/ath10k/ath10k-replace.nix:25]: Ah, damn.
Might need another update ...

Comment (ursi) [reply microsoft/surface/firmware/surface-go/ath10k/ath10k-replace.nix:25]: what part of the code is responsible for this nondeterminism?

Comment (Mic92) [reply microsoft/surface/firmware/surface-go/ath10k/ath10k-replace.nix:25]: Why does this need an outputHash anyway?

**Comment by Mic92** [comment]
Convert to draft until https://github.com/NixOS/nixos-hardware/pull/374#discussion_r801315532 is addressed.

**Pull request closed without merging**